packages/eas-cli/src/commands/update/index.ts

```diff
@@ -584,6 +571,59 @@ export default class UpdatePublish extends EasCommand {
   }
 }
 
+export async function getUpdatesToRepublishInteractiveAsync(
+  projectId: string,
+  branchName: string,
+  platformFlag: string,
+  cumulativeUpdates: Exclude<
+    Exclude<ViewBranchUpdatesQuery['app'], null | undefined>['byId']['updateBranchByName'],
+    null | undefined
+  >['updates'] = [],
+  offset: number = 0
+): Promise<any> {
+  const fetchMoreValue = '_fetchMore';
+
+  const { updates } = await ensureBranchExistsAsync({
+    appId: projectId,
+    name: branchName,
+    offset,
+  });
+  cumulativeUpdates = [...cumulativeUpdates, ...updates];
+  const updateGroups = uniqBy(cumulativeUpdates, u => u.group)
+    .filter(update => {
+      // Only show groups that have updates on the specified platform(s).
+      return platformFlag === 'all' || update.platform === platformFlag;
+    })
```

Comment:
I'm not quite sure how the old logic worked -- it seems like we need to filter by platform first, and then remove duplicates within the group. Also if platformFlag is all, we can skip the filter altogether.